libs/mimir/src/adapters/primary/common/dsl.rs

```diff
@@ -94,12 +104,117 @@ fn build_boosts(
         .clone()
         .map(|coord| build_proximity_boost(coord, &decay));
     boosts.push(proximity_boost);
-    let place_type_boost = Some(build_place_type_boost(&settings.type_query.boosts));
-    boosts.push(place_type_boost);
     boosts.into_iter().flatten().collect()
 }
 
-fn build_admin_weight_query(settings: &settings::ImportanceQueryBoosts) -> serde_json::Value {
+fn build_filters(
+    shape: Option<(Geometry, Vec<String>)>,
+    poi_types: Option<Vec<String>>,
+    zone_types: Option<Vec<String>>,
+) -> Vec<serde_json::Value> {
+    let mut filters: Vec<Option<serde_json::Value>> = Vec::new();
+    let geoshape_filter = shape.map(|(geometry, scope)| build_shape_query(geometry, scope));
+    filters.push(geoshape_filter);
+    let poi_types_filter = poi_types.map(build_poi_types_filter);
+    filters.push(poi_types_filter);
+    let zone_types_filter = zone_types.map(build_zone_types_filter);
+    filters.push(zone_types_filter);
+    filters.into_iter().flatten().collect()
```

Comment:
I think you are creating two vectors here : the first one is `filters` inside the function, and the second one is created by `collect` and returned.
I think you can avoid creating a temporary vector inside the function as follows :

```suggestion
    let mut filters: Vec<serde_json::Value> = Vec::new();
    if let Some(geoshape_filter) = shape.map(|(geometry, scope)| build_shape_query(geometry, scope)) {
        filters.push(geoshape_filter);
    };
    if let Some(poi_types_filter) = poi_types.map(build_poi_types_filter) {
        filters.push(poi_types_filter);
    }
    if let Some(zone_types_filter) = zone_types.map(build_zone_types_filter) {
        filters.push(zone_types_filter);
    }
    filters
```